docs/decisions/app/0006-use-nextjs.md

```diff
@@ -0,0 +1,101 @@
+
+## Context and Problem Statement
+
+There are many React frameworks for full-stack and front-end TypeScript/JavaScript applications. This ADR reviews several options and the pros and cons of each for the template repo.
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+### Must Haves
+- Active Maintenance: The web framework is actively maintained with patches and minor releases delivered on a regular basis
+- Community of Users: The web framework has an active community of open source users and the framework is commonly used for frontend development
+- Usability: The framework is relatively easy to learn for developers without prior experience in the particular framework, and there are plenty of resources and training materials available
+- Static Site Generation: The framework can generate static pages (HTML/CSS + JavaScript) at build time that can be cached in a CDN for faster loading
+- Server-Side Rendering: The framework can render some pages server-side with every request to get up-to-date information when the page loads
+- Client-side Rendering: The framework also supports rendering or modifying content client-side based on user interaction with the page (e.g. filtering, searching, etc.)
+
+### Nice to Have
+- AuthN/AuthZ: The framework supports Authentication & Authorization routing natively or there are established extensions that provide this functionality
+- Internationalization (i18n): The framework supports localized routing for different languages natively or there are established extensions that provide this functionality
+- Middleware: The framework supports other types of middleware (i.e. functions or scripts that execute before a routing request is complete)
+
+## Options Considered
+
+- Create React App
+- Next.js
+- Vue.js or Nuxt.js
+- Svelte or Sveltekit
+
+## Decision Outcome <!-- REQUIRED -->
+
+Chosen option: Next.js, because this option meets all our technical requirements, has a large community of support, is easy to learn with good documentation, and is well understood by Nava engineers.
+
+### Positive Consequences <!-- OPTIONAL -->
+
+- Aligning on Next.js for our template repo will help teams collaborate and share work
+
+### Negative Consequences <!-- OPTIONAL -->
+
+- We'll need to modularize our code so that if Next.js ever loses support, we can swap it out
+- While each project will determine if Next.js is a good choice, we may end up using it for projects that aren't a good fit
+
+## Pros and Cons of the Options <!-- OPTIONAL -->
+
+### Create React App
+
+Create React App is a lightweight, client side, single page application framework for React, maintained by Facebook. While it is one of the most widely adopted react frameworks, it seems like Facebook has either stopped or will stop supporting it in the near future, as it has been [removed as a suggested framework from the react website](https://github.com/reactjs/react.dev/pull/5487).
+
+- **Pros**
+  - One of the earliest react frameworks
```

Comment:
nit: but maybe better to say 'One of the earliest utilities for standing up a React application' (unless I am totally wrong, but I haven't heard create-react-app referred to as a framework before 😬 )